src/Autofac.Extensions.DependencyInjection/Autofac.Extensions.DependencyInjection.csproj

```diff
@@ -42,13 +42,13 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="1.1.0" Condition="$(TargetFramework) == 'netstandard1.1'" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.1.0" Condition="$(TargetFramework) == 'netstandard2.0'" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.0.0-preview8.19405.4" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="3.0.0-preview8.19405.4" />
```

Comment:
I will resolve this for now but still welcome comments.

Reply:
> This was not acceptable for `IWebHostBuilder` because it was specific to ASP.NET Core

So from my understanding this is not the case anymore. `Microsoft.Extensions.Hosting.Abstractions` does not contain concrete types for web-host or host. Since ASP.NET Team has built a generic-host, this can be used in full .NET-Framework / .NET-Core and ASP.NET-Core.

I definitely get the idea though 👍

Reply:
@alexmg I think having hosting stuff in here is probably not the best if we can avoid it. We had already started a small package [Autofac.AspNetCore](https://github.com/autofac/Autofac.AspNetCore) for the hosting stuff for ASP.NET Core, but since it only ever had one extension method in it, the package never got published. I think we addressed that just by putting it in the docs - here's what you do for Autofac integration.

It's certainly _prettier_ to have the extension method, but again it appears it's just a one-liner:

`hostBuilder.UseServiceProviderFactory(new AutofacServiceProviderFactory(configurationAction));`

This could be addressed by putting that in the docs, or, should we want a package for the extension method, I could see Autofac.Extensions.Hosting being something. I'd probably lean toward just putting it in the docs, though, since once we post a package it's there forever.